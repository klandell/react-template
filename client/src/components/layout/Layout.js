import React, { Component } from 'react';
import { func } from 'prop-types';

export default class Layout extends Component {
    static propTypes = {
        onAboutClick: func.isRequired,
        onNotFoundClick: func.isRequired,
    }
    render() {
        const { children, onAboutClick, onNotFoundClick } = this.props;

        return (
            <div className="layout">
                <button className="about-btn" onClick={onAboutClick}>About</button>
                <button className="not-found-btn" onClick={onNotFoundClick}>404 Not Found</button>
                {children}
            </div>
        );
    }
}
