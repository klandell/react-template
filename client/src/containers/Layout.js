import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { showAbout, showNotFound } from 'Actions/layoutActions';

@connect(state => state, dispatch => ({
    actions: {
        showAbout,
        showNotFound,
    },
    dispatch,
}))
class Layout extends React.Component {
    static propTypes = {
        actions: PropTypes.shape({
            showAbout: PropTypes.func.isRequired,
            showNotFound: PropTypes.func.isRequired,
        }).isRequired,
        dispatch: PropTypes.func.isRequired,
    }

    onAboutClick() {
        const { actions, dispatch } = this.props;
        dispatch(actions.showAbout());
    }

    onNotFoundClick() {
        const { actions, dispatch } = this.props;
        dispatch(actions.showNotFound());
    }

    render() {
        return (
            <div className="layout-wrapper">
                <button onClick={e => this.onAboutClick(e)}>About</button>
                <button onClick={e => this.onNotFoundClick(e)}>404 Not Found</button>
                {this.props.children}
            </div>
        );
    }
}
export default withRouter(Layout);
