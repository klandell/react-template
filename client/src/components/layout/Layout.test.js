import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Layout from './Layout';

const Page = {
    aboutBtn: '.about-btn',
    notFoundBtn: '.not-found-btn',
};

test('Layout component should be able to render no children.', () => {
    const component = shallow(
        <Layout
          onAboutClick={() => {}}
          onNotFoundClick={() => {}}
        />,
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
});

test('Layout component should be able to render children.', () => {
    const component = shallow(
        <Layout
          onAboutClick={() => {}}
          onNotFoundClick={() => {}}
        >
            <div>Hello</div>
            <div>World</div>
        </Layout>,
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
});

test('Layout component should call props.onAboutClick when the about button is clicked.', () => {
    const onAboutClickMock = jest.fn();
    const component = shallow(
        <Layout
          onAboutClick={onAboutClickMock}
          onNotFoundClick={() => {}}
        />,
    );

    component.find(Page.aboutBtn).simulate('click');
    expect(onAboutClickMock).toHaveBeenCalled();
});

test('Layout component should call props.onNotFoundClick when the not found button is clicked.', () => {
    const onNotFoundClickMock = jest.fn();
    const component = shallow(
        <Layout
          onAboutClick={() => {}}
          onNotFoundClick={onNotFoundClickMock}
        />,
    );

    component.find(Page.notFoundBtn).simulate('click');
    expect(onNotFoundClickMock).toHaveBeenCalled();
});
