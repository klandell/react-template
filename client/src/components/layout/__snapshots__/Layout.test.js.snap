// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should be able to render children. 1`] = `
<div
  className="layout"
>
  <button
    className="about-btn"
    onClick={[Function]}
  >
    About
  </button>
  <button
    className="not-found-btn"
    onClick={[Function]}
  >
    404 Not Found
  </button>
  <div>
    Hello
  </div>
  <div>
    World
  </div>
</div>
`;

exports[`Layout component should be able to render no children. 1`] = `
<div
  className="layout"
>
  <button
    className="about-btn"
    onClick={[Function]}
  >
    About
  </button>
  <button
    className="not-found-btn"
    onClick={[Function]}
  >
    404 Not Found
  </button>
</div>
`;
